lib/src/common/repos_service.dart

```diff
@@ -469,71 +684,236 @@ class RepositoriesService extends Service {
+  /// * [name]: The name of the release.
+  /// * [body]: Text describing the contents of the tag.
+  /// * [draft]: true makes the release a draft, and false publishes the release.
+  /// * [preRelease]: true to identify the release as a prerelease, false to identify the release as a full release.
+  ///
+  /// Leave blank the fields you don't want to edit.
+  ///
+  /// API docs: https://developer.github.com/v3/repos/releases/#edit-a-release
+  Future<Release> editRelease(
+    RepositorySlug slug,
+    Release releaseToEdit, {
+    String tagName,
+    String targetCommitish,
+    String name,
+    String body,
+    bool draft,
+    bool preRelease,
+  }) async {
+    return _github.postJSON<Map<String, dynamic>, Release>(
+      "/repos/${slug.fullName}/releases/${releaseToEdit.id.toString()}",
+      body: jsonEncode(createNonNullMap(<String, dynamic>{
+        "tag_name": tagName ?? releaseToEdit.tagName,
+        "target_commitish": targetCommitish ?? releaseToEdit.targetCommitish,
+        "name": name ?? releaseToEdit.name,
+        "body": body ?? releaseToEdit.body,
+        "draft": draft ?? releaseToEdit.draft,
+        "prerelease": preRelease ?? releaseToEdit.prerelease,
+      })),
+      statusCode: StatusCodes.OK,
+      convert: (i) => Release.fromJson(i),
+    );
+  }
+
+  /// Delete the release.
+  ///
+  /// API docs: https://developer.github.com/v3/repos/releases/#delete-a-release
+  Future<void> deleteRelease(RepositorySlug slug, Release release) async {
+    await _github.request(
+      "DELETE",
+      "/repos/${slug.fullName}/releases/${release.id.toString}",
+      statusCode: StatusCodes.NO_CONTENT,
+    );
+  }
+
+  /// Lists assets for a release.
+  ///
+  /// API docs: https://developer.github.com/v3/repos/releases/#list-assets-for-a-release
+  Stream<ReleaseAsset> listReleaseAssets(RepositorySlug slug, Release release) {
+    return PaginationHelper(_github)
+        .objects<Map<String, dynamic>, ReleaseAsset>(
+      "GET",
+      "/repos/${slug.fullName}/releases/${release.id.toString()}/assets",
+      (i) => ReleaseAsset.fromJson(i),
+      statusCode: StatusCodes.OK,
+    );
+  }
+
+  /// Get a single release asset.
+  ///
+  /// API docs: https://developer.github.com/v3/repos/releases/#get-a-single-release-asset
+  // TODO: implement a way to retrieve the asset's binary content
+  Future<ReleaseAsset> getReleaseAsset(RepositorySlug slug, Release release,
+      {@required int assetId}) async {
+    return _github.postJSON<Map<String, dynamic>, ReleaseAsset>(
+      "/repos/${slug.fullName}/releases/assets/${assetId.toString()}",
+      statusCode: StatusCodes.OK,
+      convert: (i) => ReleaseAsset.fromJson(i),
+    );
+  }
+
+  /// Edits a release asset.
+  ///
+  /// API docs: https://developer.github.com/v3/repos/releases/#edit-a-release-asset
+  Future<ReleaseAsset> editReleaseAsset(
+    RepositorySlug slug,
+    ReleaseAsset assetToEdit, {
```

Comment:
You might want to protect against if the user passes in something null here.

Reply:
I added null checks for all methods in this file